packages/wallet-sdk/src/derive.ts

```diff
@@ -248,25 +278,45 @@ export const deriveStxPrivateKey = ({
   rootNode,
   index,
 }: {
-  rootNode: BIP32Interface;
+  rootNode: HDKey | BIP32Interface; // Backward compatibility with bip32
   index: number;
 }) => {
-  const childKey = rootNode.derivePath(STX_DERIVATION_PATH).derive(index);
+  let childKey;
+  if (rootNode instanceof HDKey) {
+    childKey = rootNode.derive(STX_DERIVATION_PATH).deriveChild(index);
+  } else {
+    // Backward compatibility with bip32
+    childKey = rootNode.derivePath(STX_DERIVATION_PATH).derive(index);
+  }
   assertIsTruthy(childKey.privateKey);
-  const ecPair = ECPair.fromPrivateKey(childKey.privateKey);
+  const privateKey =
+    childKey.privateKey instanceof Uint8Array
+      ? Buffer.from(childKey.privateKey)
+      : childKey.privateKey;
+  const ecPair = ECPair.fromPrivateKey(privateKey);
   return ecPairToHexString(ecPair);
 };
 
 export const deriveDataPrivateKey = ({
   rootNode,
   index,
 }: {
-  rootNode: BIP32Interface;
+  rootNode: HDKey | BIP32Interface; // Backward compatibility with bip32
   index: number;
 }) => {
-  const childKey = rootNode.derivePath(DATA_DERIVATION_PATH).deriveHardened(index);
+  let childKey;
+  if (rootNode instanceof HDKey) {
+    childKey = rootNode.derive(DATA_DERIVATION_PATH).deriveChild(index + HIGHEST_BIT);
```

Comment:
Why do we need `HIGHEST_BIT` is some cases and not others? Where does this come from?

Reply:
`+ HIGHEST_BIT` "hardens", right? Should we be doing this everywhere?

Reply:
Dismiss my previous comments. But let's rename `HIGHEST_BIT` to something like `HARDENED_OFFSET` to make it more clear

Reply:
✅